GitAutomation/GitAutomation.psm1

```diff
@@ -26,7 +26,12 @@ if( $gitCmd )
 
 if( -not $registeredSsh )
 {
-    Write-Warning -Message ('SSH support is disabled. To enable SSH, please install Git for Windows. GitAutomation uses the version of SSH that ships with Git for Windows.')
+    $local:Message = 'SSH support is disabled. To enable SSH, please install Git for Windows. GitAutomation uses the version of SSH that ships with Git for Windows.'
+    if ($env:AUTOMATION_ASSET_ENDPOINT) {
```

Comment:
This change doesn't solve the problem, rather it just suppresses the warning in a very specific one-off scenario. The proper solution is to make the logic more robust to better handle when the warning should and shouldn't be written. But also this warning is bound to be rewritten/removed when we work to migrate this module to support running on PowerShell Core and MacOS/Linux anyways.

If you want to just suppress this warning in your specific case with that environment variable, this logic should be one level up in the script/module that imports this module. You can use [redirection](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_redirection?view=powershell-5.1) to redirect the warning stream (`3`) to null like so:

```
Import-Module -Name 'GitAutomation' 3> $null
```

Reply:
I see - but if I redirect the whole warning stream to `$null`, I wouldn't get any other warnings as well. What about a parameter (e.g. `-NoSshCheck` or `-SkipSshCheck`) to skip the entire check? That way, the logic can be moved one level up, as you suggested, while retaining any other warnings that might come up (also in the future). Also, if this check will actually get removed, the parameter can be deprecated by issuing another warning message `The xxx parameter is deprecated ...` if the parameter has been specified. so those who don't want unsolicited warnings in their logs will be eager to remove the parameter as well.

Reply:
Since this is on module import, I don't want to add a parameter. I'm not sure people know you can pass arguments to a module when importing it?

What problem are you trying to solve? Why do you want to suppress the warning?

Reply:
I didn't know either, but stackoverflow.com almost always helps ;)

We're using GitAutomation in Azure Automation Runbooks to fetch files from a git repo. We heavily use Log Analytics to provide ourselves with proper monitoring - we need to know about every warning and every error. While Azure Automation has only rough Job status inormation (Completed, Failed or Suspended, but nothing like 'HadWarnings' or 'HadNonTerminatingErrors'), Log Analytics provides ways to inspect the detailed Job logs, which are made up, amongst others, of everything you emit through the `Write-*` cmdlets. So, we use those to clearly indicate if everything is fine - if there are any warning- or error-type log entries, the overall Job result will be the highest severity found in any log record. Thus, any jobs using GitAutomation would never get a 'Success' status, but always 'Warning' only. This forces us to browse the logs after each and every job run to check if the "Warning" status was caused by anything else than the "SSH support is disabled" message.

The message pretends a dependency that doesn't exist if you don't use SSH. We're using GitAutomation to avoid the need to install git in the Runbook Workers every time we start those jobs (if this can be done at all, I didn't even try it).

To turn such parameter into something more meaningful than just avoiding a message, it could also be `-DisableSsh` and provide the functionality to intentionally unregister `ssh.exe` - for whichever reason that might be useful …

I hope this explanation helps understanding what we're trying to achieve.

Reply:
@splatteredbits how do you want me to proceed here?

Reply:
How are you importing `GitAutomation`? Can't you run `Import-Module -Name 'GitAutomation' 3> $null`  wherever you're importing to ignore the SSH error?

The correct solution to this is Pretty Big and requires a lot of changes to the module involving detecting and checking the SSH state just before pushing/pulling to a remote. Or testing what happens if SSH isn't configured and someone attempts to push/pull via SSH.

Reply:
@splatteredbits ok I'll revert this change but leave the UTF8 file encoding @DecoyJoe recommended for future use. Currently I have no idea how to make the correct solution needing less effort, so I'll just create a new issue we can continue working on and maintain the proposed changes in the fork.